File: 백준/Bronze/2884. 알람 시계/알람 시계.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.StringTokenizer

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val stringTokenizer = StringTokenizer(readLine(), " ")
    var hour = stringTokenizer.nextToken().toInt()
    val minute = stringTokenizer.nextToken().toInt()
    if (hour == 0) hour = 24
    var total = hour * 60 + minute - 45
    val secondTotal = total % 60

    if (total / 60 >= 24) total = 0

    print("${total / 60} $secondTotal")
}

File: 백준/Bronze/2884. 알람 시계/알람 시계.py
hour, minute = map(int, input().split())

if hour == 0:
    hour = 24

hour = (hour * 60 + minute - 45) // 60
minute = (hour * 60 + minute - 45) % 60

if hour == 24:
    print(0, minute)
else:
    print(hour, minute)